runtime/devnet/src/extensions/mod.rs

```diff
@@ -373,118 +290,27 @@ impl TryFrom<u8> for FuncId {
 	}
 }
 
-fn read_parachain_system_state<T, E>(
-	key: ParachainSystemKeys,
-	env: &mut Environment<E, BufInBufOutState>,
-) -> Result<Vec<u8>, DispatchError>
-where
-	T: pallet_contracts::Config + cumulus_pallet_parachain_system::Config,
-	E: Ext<T = T>,
-{
-	match key {
-		ParachainSystemKeys::LastRelayChainBlockNumber => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(RelaychainDataProvider::<T>::current_block_number().encode())
-		},
-	}
-}
-
-fn read_nfts_state<T, E>(
-	key: NftsKeys,
-	env: &mut Environment<E, BufInBufOutState>,
-) -> Result<Vec<u8>, DispatchError>
-where
-	T: pallet_contracts::Config + pallet_nfts::Config<CollectionId = CollectionId, ItemId = ItemId>,
-	E: Ext<T = T>,
-{
-	match key {
-		NftsKeys::Collection(collection) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(pallet_nfts::Collection::<T>::get(collection).encode())
-		},
-		NftsKeys::CollectionOwner(collection) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(pallet_nfts::Pallet::<T>::collection_owner(collection).encode())
-		},
-		NftsKeys::Item(collection, item) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(pallet_nfts::Item::<T>::get(collection, item).encode())
-		},
-		NftsKeys::Owner(collection, item) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(pallet_nfts::Pallet::<T>::owner(collection, item).encode())
-		},
-		NftsKeys::Attribute(collection, item, key) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(pallet_nfts::Pallet::<T>::attribute(&collection, &item, &key).encode())
-		},
-		// NftsKeys::CustomAttribute(account, collection, item, key) => {
-		// 	env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-		// 	Ok(pallet_nfts::Pallet::<T>::custom_attribute(&account, &collection, &item, &key)
-		// 		.encode())
-		// },
-		NftsKeys::SystemAttribute(collection, item, key) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(pallet_nfts::Pallet::<T>::system_attribute(&collection, item.as_ref(), &key)
-				.encode())
-		},
-		NftsKeys::CollectionAttribute(collection, key) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(pallet_nfts::Pallet::<T>::collection_attribute(&collection, &key).encode())
-		},
-	}
-}
-
-fn read_assets_state<T, E>(
-	key: AssetsKeys,
+fn read_fungibles_state<T, E>(
```

Comment:
Replacing all the individual read functions with pallet::read(key) could eliminate the need for this function.

Reply:
I will show #101 prototype with a PR to this branch and see if we want that. If so, this will not be possible because we need to return the result directly with its type in stead of encoding it.